tests/aws/test_s3.py

```diff
@@ -120,8 +120,10 @@ def test_should__upload_dataframe_and_download_parquet__when_using_s3util(self):
         upload_key = "temp456"
         test_object = pd.DataFrame([1, 2, 3, 4], columns=["one"])
         self.s3.upload_dataframe_as_parquet(dataframe=test_object, key=upload_key)
-        redown_df = self.s3.download_parquet_as_dataframe(f"{upload_key}/data.parquet")
-        assert_frame_equal(test_object, redown_df)
+        self.s3.download_file(local_file_path="./test_parquet_file.parquet", key=upload_key + '/')
```

Comment:
why are you doing this?

Reply:
what was the need for this change?

Reply:
the reason why you are getting a 404 not found is because ```self.s3.download_file(local_file_path="./test_parquet_file.parquet", key=upload_key + '/')```

points to a directory not a file in s3, so it is not a valid key, hence not found.

Reply:
I have changed this test friday to track some other behaviour. As you said it was just a path issue. Now it is working.